lib/acts_as_ar_query.rb

```diff
@@ -194,6 +194,35 @@ def append_hash_arg(symbol, *val)
     end
   end
 
+  def append_hash_array_arg(symbol, default, *val)
+    val = val.flatten.compact
+    val = val.first if val.size == 1 && val.first.kind_of?(Hash)
```

Comment:
Looking at docs [here](http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-where)

Extracting the hash makes options like `{}` vs `["",{}]` too confusing for me.

Think we just want the case to say, if they are passing in only a hash, then get rid of the outer array that was added by the splat operator.

And the flatten gets rid of a double array added by the splat operator